src/main/service.rs

```diff
@@ -196,6 +196,11 @@ impl<UID: Uid> Service<UID> {
         }
     }
 
+    /// Return the ip address of the peer.
+    pub fn get_peer_ip_addr(&self, peer_id: &UID) -> ::Res<IpAddr> {
+        Ok(unwrap!(self.get_peer_socket_addr(peer_id)).ip())
```

Comment:
why are we using unwrap in production code when it's not used anywhere else in the call to `get_peer_socket_addr()` - why not simply `self.get_peer_socket_addr(...).map(|s| s.ip())` ? ( <-- if using that then pls copy exactly that as it's a convention elsewhere too to call socket-addr `s` inside closures)